### PR TITLE
Coff line number

### LINKED ISSUE #55: Support COFF line numbers
MSVC stores line numbers in obj as extra data accessed via a pointer in various headers.
Official specs are available here "5.3. COFF Line Numbers"
https://courses.cs.washington.edu/courses/cse378/03wi/lectures/LinkerFiles/coff.pdf
Pointers are mentioned in
"4. Section Table (Section Headers)"
"5.5.1. Auxiliary Format 1: Function Definitions"

### PR DESCRIPTION
**Pull request by roblabla**
Fixes #55 

This turned out to be fairly straightforward once I figured out how on earth those line numbers _really_ worked. I tried to comment everything so it should be clear what's going on, but if you need more information or something feels unclear, feel free to ask and I'll see if I can improve the comments to make it clearer.

### CODE REVIEW

File: objdiff-core/src/obj/read.rs [hunk 1]
```diff
@@ -490,6 +493,110 @@ fn line_info(obj_file: &File<'_>, sections: &mut [ObjSection]) -> Result<()> {
...
+
+    Ok(())
+}
+
+fn line_info_coff(coff: &CoffFile, sections: &mut [ObjSection], obj_data: &[u8]) -> Result<()> {
+    let symbol_table = coff.coff_header().symbols(obj_data)?;
+
+    // Enumerate over all sections.
+    for sect in coff.sections() {
+        let ptr_linenums = sect.coff_section().pointer_to_linenumbers.get(LE) as usize;
```
Comment (encounter) [inline objdiff-core/src/obj/read.rs:509]: I'm guessing we don't need to care about BE COFF? :slightly_smiling_face: I imagine they're pretty rare

**Review by encounter — COMMENTED** [review]
Thanks so much for the impl! I wasn't really looking forward to figuring this out on my own. :laughing: Code looks great, and I appreciate the documentation.

I use nightly fmt options so you can do `cargo +nightly fmt` to resolve the issues.

File: objdiff-core/src/obj/read.rs (hunk 1 above)
Comment (roblabla) [reply objdiff-core/src/obj/read.rs:509]: COFFs are always LE (you can't actually pass BE here - Object expects you to pass the LE object, as [`pointer_to_linenumbers` is of type `U32Bytes<LittleEndian>`](https://docs.rs/object/latest/object/pe/struct.ImageAuxSymbolFunction.html#structfield.pointer_to_linenumber). The whole "pass endian object to value" in object is a bit weird...)

**Comment by roblabla** [comment]
Formatting should be fixed.

**Pull request merged**